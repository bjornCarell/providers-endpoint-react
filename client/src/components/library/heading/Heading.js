import React from 'react';
import PropTypes from 'prop-types';
import {
  HeadingOne,
  HeadingTwo,
  HeadingThree,
  HeadingFour,
  HeadingFive
} from './HeadingStyled';

export const Heading = ({
  h2,
  h3,
  h4,
  h5,
  noMargin,
  right,
  center,
  ...props
}) => {
  if (h2)
    return (
      <HeadingTwo
        noMargin={noMargin}
        right={right}
        center={center}
        {...props}
      />
    );
  if (h3)
    return (
      <HeadingThree
        noMargin={noMargin}
        right={right}
        center={center}
        {...props}
      />
    );
  if (h4)
    return (
      <HeadingFour
        noMargin={noMargin}
        right={right}
        center={center}
        {...props}
      />
    );
  if (h5)
    return (
      <HeadingFive
        noMargin={noMargin}
        right={right}
        center={center}
        {...props}
      />
    );
  return (
    <HeadingOne noMargin={noMargin} right={right} center={center} {...props} />
  );
};

Heading.propTypes = {
  h2: PropTypes.bool,
  h3: PropTypes.bool,
  h4: PropTypes.bool,
  h5: PropTypes.bool,
  noMargin: PropTypes.bool,
  right: PropTypes.bool,
  center: PropTypes.bool
};
