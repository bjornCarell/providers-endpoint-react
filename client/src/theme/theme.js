export const theme = {
  color: {
    primary: '#004146',
    primaryLight: '#427783',
    primaryBright: '#E8F9F8',
    secondary: '#F89572',
    secondaryLight: '#FDDFD4',
    secondaryDark: '#FF8A61',
    secondaryBright: '#FFE8E5',
    white: '#FFFFFF',
    light: '#fcfcfc',
    black: '#262626',
    grey: '#F5F6F7'
  },
  spacing: {
    xss: '0.5rem',
    xs: '0.75rem',
    small: '1rem',
    medium: '2rem',
    large: '4rem',
    xl: '8rem',
    xxl: '16rem'
  },
  fontSize: {
    small: '1.2rem',
    medium: '1.6rem',
    large: '2.6rem',
    xl: '4.22rem',
    xxl: '6.85rem'
  },
  fontWeight: {
    small: 200,
    medium: 400,
    large: 600,
    xl: 800
  },
  animation: {
    shadowAround: '0px 0px 10px 0px rgba(0,0,0,0.1)',
    shadowAroundHover: '0px 0px 16px 0px rgba(0,0,0,0.13)',
    shadowBottom: '0px 4px 8px 0 rgba(0,0,0,0.1)',
    shadowBottomHover: '0px 8px 8px 0 rgba(0,0,0,0.12)',
    radiusS: '0.375rem',
    radiusM: '0.875rem',
    radiusL: '2rem',
    transition: 'all .25s cubic-bezier(0.4,0,0.2,1)'
  },
  zIndex: {
    sidebar: 4
  }
};
