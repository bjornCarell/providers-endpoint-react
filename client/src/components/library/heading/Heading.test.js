import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../../theme/theme';
import { Heading } from './Heading';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(
    <ThemeProvider theme={theme}>
      <Heading theme={theme} />
    </ThemeProvider>,
    div
  );
});

it('renders h1 with conditional style', () => {
  const tree = renderer
    .create(<Heading theme={theme}>Providers</Heading>)
    .toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', `${theme.fontSize.xxl}`);
  expect(tree).toHaveStyleRule('margin-bottom', `${theme.spacing.large}`);
});

it('renders h1 with no margin and text-align center', () => {
  const tree = renderer
    .create(
      <Heading theme={theme} noMargin center>
        Providers
      </Heading>
    )
    .toJSON();

  expect(tree).toHaveStyleRule('margin-bottom', '0');
  expect(tree).toHaveStyleRule('text-align', 'center');
});

it('renders h1 with text-align right', () => {
  const tree = renderer
    .create(
      <Heading theme={theme} right>
        Providers
      </Heading>
    )
    .toJSON();

  expect(tree).toHaveStyleRule('text-align', 'right');
});

it('renders h2 with conditional style', () => {
  const tree = renderer
    .create(
      <Heading theme={theme} h2>
        Providers
      </Heading>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', `${theme.fontSize.xl}`);
  expect(tree).toHaveStyleRule('margin-bottom', `${theme.spacing.medium}`);
});

it('renders h3 with conditional style', () => {
  const tree = renderer
    .create(
      <Heading theme={theme} h3>
        Providers
      </Heading>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', `${theme.fontSize.large}`);
  expect(tree).toHaveStyleRule('margin-bottom', `${theme.spacing.small}`);
});

it('renders h3 with conditional style', () => {
  const tree = renderer
    .create(
      <Heading theme={theme} h4>
        Providers
      </Heading>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', `${theme.fontSize.medium}`);
  expect(tree).toHaveStyleRule('margin-bottom', `${theme.spacing.xs}`);
});

it('renders h5 with conditional style', () => {
  const tree = renderer
    .create(
      <Heading theme={theme} h5>
        Providers
      </Heading>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', `${theme.fontSize.small}`);
  expect(tree).toHaveStyleRule('margin-bottom', `${theme.spacing.xss}`);
});
