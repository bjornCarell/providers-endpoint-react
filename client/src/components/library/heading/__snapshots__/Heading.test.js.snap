// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders h1 with conditional style 1`] = `
.c0 {
  font-size: 6.85rem;
  margin-bottom: 4rem;
  color: #262626;
  font-weight: 600;
  margin-top: 0;
  max-width: 100%;
  text-align: start;
}

<h1
  className="c0"
>
  Providers
</h1>
`;

exports[`renders h2 with conditional style 1`] = `
.c0 {
  font-size: 4.22rem;
  margin-bottom: 2rem;
  color: #262626;
  font-weight: 600;
  margin-top: 0;
  max-width: 100%;
  text-align: start;
}

<h2
  className="c0"
>
  Providers
</h2>
`;

exports[`renders h3 with conditional style 1`] = `
.c0 {
  font-size: 2.6rem;
  margin-bottom: 1rem;
  color: #262626;
  font-weight: 600;
  margin-top: 0;
  max-width: 100%;
  text-align: start;
}

<h3
  className="c0"
>
  Providers
</h3>
`;

exports[`renders h3 with conditional style 2`] = `
.c0 {
  font-size: 1.6rem;
  margin-bottom: 0.75rem;
  color: #262626;
  font-weight: 600;
  margin-top: 0;
  max-width: 100%;
  text-align: start;
}

<h4
  className="c0"
>
  Providers
</h4>
`;

exports[`renders h5 with conditional style 1`] = `
.c0 {
  font-size: 1.2rem;
  margin-bottom: 0.5rem;
  color: #262626;
  font-weight: 600;
  margin-top: 0;
  max-width: 100%;
  text-align: start;
}

<h5
  className="c0"
>
  Providers
</h5>
`;
